src/pcr-policy.c

```diff
@@ -40,6 +40,27 @@
 #include "tpm.h"
 #include "config.h"
 #include "tpm2key.h"
+#include "sd-boot.h"
+
+struct target_platform {
+        const char *    name;
+	unsigned int	unseal_flags;
+
+        bool            (*write_sealed_secret)(const char *pathname,
+					const TPML_PCR_SELECTION *pcr_sel,
+					const TPM2B_PRIVATE *sealed_private,
+					const TPM2B_PUBLIC *sealed_public);
+        bool            (*write_signed_policy)(const char *input_path, const char *output_path,
```

Comment:
Maybe at this stage we can think about this interface.

As commented I plan to add multiple policies in the JSON format, that imply that we need to pass multiple `TPM2B_DIGEST` (and I guess multiple `TPMT_SIGNATURE`), or an extra parameter `const bool append` to avoid the replacement of the full file

Reply:
Well, for tpm2.0 keys, the default behavior is currently

- read from input
- add the new policy
- write to output

We could do the same for the systemd case. I think it may be better to always update in-place instead. It would be the job of sdb_policy_file_add_entry() to open the json file, add a new entry, and write it back

Reply:
>  It would be the job of sdb_policy_file_add_entry() to open the json file, add a new entry, and write it back

Yes, I see: whould be possible to call multiple times `platform->write_signed_policy()` to add new entries. In that case what is missing is some reset of the json file. Would be OK a `--wipe` parameter to `sign` to remove the output file before adding entries?

Reply:
Actually, I would prefer if we do not make pcr-oracle too powerful. Instead, I'd create one or more scripts that understand all the sd-boot specifics and call pcr-oracle only for the essential stuff.

So instead of a --wipe option, I would do a "rm -f blah.json" in that wrapper script. Similarly, I wouldn't add code that makes pcr-oracle loop over all sd-boot entries, but do a shell loop instead:

 for conf in /efi/boot/loader/*.conf; do
  id=$(basename "$conf" .conf)
  pcr-oracle --next-kernel $id ...
done

Reply:
OK, so is "always append".

Do you know if the TPM2 and oldgrub formats support multiple predictions in the same struct?

Reply:
TPM2.0 keys yes, oldgrub no.